enterprise/server/backends/authdb/authdb.go

```diff
@@ -180,16 +189,32 @@ func (d *AuthDB) LookupUserFromSubID(ctx context.Context, subID string) (*tables
 	return user, nil
 }
 
-func newAPIKeyGroupQuery() *query_builder.Query {
+func newAPIKeyGroupQuery(allowUserOwnedKeys bool) *query_builder.Query {
 	qb := query_builder.NewQuery(`
 		SELECT
 			ak.capabilities,
+			ak.user_id,
 			g.group_id,
 			g.use_group_owned_executors
 		FROM ` + "`Groups`" + ` AS g,
 		APIKeys AS ak
 	`)
 	qb.AddWhereClause(`ak.group_id = g.group_id`)
+	if allowUserOwnedKeys {
+		// Note: the org can disable user-owned keys at any time, and the
+		// predicate here ensures that existing keys are effectively deactivated
+		// (but not deleted).
+		qb.AddWhereClause(`(
```

Comment:
Should we change `AddWhereClause` to automatically add parenthesis around the clause? Expecting users to remember to do it seems like asking for trouble.

Reply:
Seems like a good idea to me. I can send a separate PR for that

Reply:
+1

Reply:
Sent https://github.com/buildbuddy-io/buildbuddy/pull/3444/files